gc3libs/workflow.py

```diff
@@ -401,7 +401,10 @@ def redo(self, from_stage=0, *args, **kwargs):
             task.redo(*args, **kwargs)
         # All other tasks should be put in NEW again
         for i in range(from_stage+1, len(self.tasks)):
-            self.tasks[i].execution.state = Run.State.NEW
+            try:
+                self.tasks[i].redo(*args, **kwargs)
+            except IndexError:
```

Comment:
...but this looks to me more like a bug in SequentialTaskCollection.stage(): `self._current_task` has an invalid value.

Can you please add a line before line 443::

    assert (0 <= self._current_task < len(self.tasks)), ("Invalid value self._current_task=%d in %s" % (self._current_task, self))

run again and print the traceback and error message?